source/AI.cpp

```diff
@@ -2810,56 +2948,105 @@ bool AI::DoSecretive(Ship &ship, Command &command)
 
 
 // Instead of coming to a full stop, adjust to a target velocity vector
-Point AI::StoppingPoint(const Ship &ship, const Point &targetVelocity, bool &shouldReverse)
+Point AI::StoppingPoint(const Ship &ship, const Point &targetVelocity, bool reverse, double &timeToStop, Point &toFace)
 {
 	Point position = ship.Position();
-	Point velocity = ship.Velocity() - targetVelocity;
+	Point velocity = ship.Velocity();
 	Angle angle = ship.Facing();
-	double acceleration = ship.Acceleration();
+	double acceleration;
+	if(!reverse)
+		acceleration = ship.Acceleration();
+	else
+		acceleration = ship.Attributes().Get("reverse thrust") / ship.InertialMass();
+
 	double turnRate = ship.TurnRate();
-	shouldReverse = false;
 
 	// If I were to turn around and stop now the relative movement, where would that put me?
-	double v = velocity.Length();
-	if(!v)
+
+	double v = (velocity - targetVelocity).Length();
+	if(v < 1e-6)
```

Comment:
You use this `1e-6` a few times, maybe put it in a `static const` variable at the top of this method?
Also, I think it should be `1.e-6`, for consistency?